framework/include/lightsource.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+namespace cgb
+{
+	/**	Lightsource struct which hold all the concrete data,
+	*/
+	struct lightsource
+	{
+		std::string mName;
+
+		float mAngleInnerCone;
+		float mAngleOuterCone;
+		float mAttenuationConstant;
+		float mAttenuationLinear;
+		float mAttenuationQuadratic;
+		glm::vec3 mColorAmbient;
+		glm::vec3 mColorDiffuse;
+		glm::vec3 mColorSpecular;
+		glm::vec3 mDirection;
+		glm::vec3 mPosition;
+		int mType;
```

Comment:
Please use an enum struct for the type => `enum struct lightsource_type { directional, point, spot }`